presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -169,12 +187,30 @@ private FileSystem getCachingFileSystem(
     public void tearDown()
             throws IOException
     {
+        nonCachingFileSystem.close();
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void closeRubix()
+            throws IOException
+    {
+        if (cachingFileSystem == null) {
```

Comment:
This should be part of lambda passed to `closer.register` below.

Reply:
Added explicit `cachingInitialized` field